### PR TITLE
Outlaw Polish

### LINKED ISSUE #4067: Rolling the bones not being recommended properly
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Spec

Rogue - Outlaw

### Describe the Issue

Just one example, I have both broadside and true bearing active. RTB should not be recommended here especially since loaded dice is also not active as well.

### How to Reproduce

Press RTB and snapshot

### Snapshot (Link)

https://pastebin.com/RJZGvrU8

### Raidbots Sim Report (Link)

_No response_

### Additional Information

_No response_

### Contact Information

_No response_

### PR DESCRIPTION
**Pull request by syrifgit**
- Remove unnecessary spec options & references in APL and class file
  - These options should not be user options, they are all handled by the APL and no nothing but lower DPS
  - Stealth-bug prevention options also no longer needed since BTE CD is refreshed upon stealthing 
- Change Adrenaline Rush stealth-break behaviour to account for the current fatebound bug
- Probably put poisons on before you activate a stat buff trinket
- Update pack string
- fix #4067 RTB variable

### CODE REVIEW

File: TheWarWithin/Priorities/RogueOutlaw.simc [hunk 1]
```diff
@@ -75,7 +75,7 @@ actions.cds+=/call_action_list,name=stealth_cds,if=!stealthed.all
 actions.cds+=/thistle_tea,if=!buff.thistle_tea.up&(energy.base_deficit>=150|fight_remains<charges*6)
 # Use Blade Rush at minimal energy outside of stealth
 actions.cds+=/blade_rush,if=energy.base_time_to_max>4&!stealthed.all
-actions.cds+=/potion,if=buff.bloodlust.react|fight_remains<30|buff.adrenaline_rush.up
+actions.cds+=/potion,if=buff.bloodlust.up|fight_remains<30|buff.adrenaline_rush.up
```
Comment (Hekili) [inline TheWarWithin/Priorities/RogueOutlaw.simc]: Let's do `boss&fight_remains<30` so we're not blowing potion CD on every short-lived pack.

File: TheWarWithin/RogueOutlaw.lua [hunk 2]
```diff
@@ -891,7 +891,7 @@ spec:RegisterAbilities( {
         gcd = "off",
 
         talent = "adrenaline_rush",
-        startsCombat = false,
+        startsCombat = function () return talent.hand_of_fate.enabled and true or false end,
```
Comment (Hekili) [inline TheWarWithin/RogueOutlaw.lua]: `function() return talent.hand_of_fate.enabled end,`

**Review by Hekili — COMMENTED** [review]
Looks mostly fine; could just delete the obsolete settings instead of leaving comments but nbd either way.

Not merging yet; looking into making the rtb_reroll function without moving further from SimC.

File: TheWarWithin/Priorities/RogueOutlaw.simc [hunk 3]
```diff
@@ -13,7 +13,7 @@ actions+=/kick
...
 actions+=/variable,name=rtb_reroll,value=rtb_buffs.will_lose=(rtb_buffs.will_lose.buried_treasure+rtb_buffs.will_lose.grand_melee&spell_targets.blade_flurry<2)
 # If Loaded Dice is talented, then keep any 1 buff from Roll the Bones but roll it into 2 buffs when Loaded Dice is active. Also reroll 2 buffs with loaded dice up if broadside, ruthless precision and true bearing are all missing and loaded dice is up
-actions+=/variable,name=rtb_reroll,if=talent.loaded_dice,value=(rtb_buffs.will_lose<=buff.loaded_dice.up)|buff.loaded_dice.up&rtb_buffs.will_lose<3&(!rtb_buffs.will_lose.broadside|buff.broadside.remains<11)&(!rtb_buffs.will_lose.ruthless_precision|buff.ruthless_precision.remains<11)&(!rtb_buffs.will_lose.true_bearing|buff.true_bearing.remains<11)
+actions+=/variable,name=rtb_reroll,if=talent.loaded_dice,value=((rtb_buffs.will_lose<=1&buff.loaded_dice.up)|(rtb_buffs.will_lose<1&!buff.loaded_dice.up))|buff.loaded_dice.up&rtb_buffs.will_lose<3&(!rtb_buffs.will_lose.broadside|buff.broadside.remains<11)&(!rtb_buffs.will_lose.ruthless_precision|buff.ruthless_precision.remains<11)&(!rtb_buffs.will_lose.true_bearing|buff.true_bearing.remains<11)
```
Comment (Hekili) [inline TheWarWithin/Priorities/RogueOutlaw.simc]: For expediency, I would merge:

> actions+=/variable,name=rtb_reroll,if=talent.loaded_dice,value=`(rtb_buffs.will_lose<=1*buff.loaded_dice.up)`|buff.loaded_dice.up&rtb_buffs.will_lose<3&(!rtb_buffs.will_lose.broadside|buff.broadside.remains<11)&(!rtb_buffs.will_lose.ruthless_precision|buff.ruthless_precision.remains<11)&(!rtb_buffs.will_lose.true_bearing|buff.true_bearing.remains<11)

This converts internally to `( rtb_buffs_will_lose <= 1 * safenum( buff.loaded_dice.up ) )`, where `safenum` converts a boolean ( `false`/`nil` -> `0`; `true` -> `1` ).

**Review by Hekili — CHANGES_REQUESTED** [review]
Simpler change for expediency.

File: TheWarWithin/Priorities/RogueOutlaw.simc (hunk 3 above)
Comment (syrifgit) [reply TheWarWithin/Priorities/RogueOutlaw.simc]: Awesome, will get a new pack string for that version.

Comment (syrifgit) [reply TheWarWithin/Priorities/RogueOutlaw.simc]: Works as intended
```
        10.  variable ( default - 4 )
        rtb_reroll #2 [Outlaw:default:4]; conditions = PASS: talent.loaded_dice.enabled[true]
         - value = false: ( rtb_buffs_will_lose[2.00] <= 1 * buff.loaded_dice.up[false] ) | buff.loaded_dice.up[false] & rtb_buffs_will_lose[2.00] < 3 & ( ! rtb_buffs_will_lose_buff.broadside[false] | buff.broadside.remains[0.00] < 11 ) & ( ! rtb_buffs_will_lose_buff.ruthless_precision[false] | buff.ruthless_precision.remains[0.00] < 11 ) & ( ! rtb_buffs_will_lose_buff.true_bearing[true] | buff.true_bearing.remains[27.47] < 11 )
         - variable.rtb_reroll[false] will check this script entry ( Outlaw:default:4 )
        Time spent on this action:  0.03ms
        TimeData:Outlaw-default-4:variable:0.03:Ability Known, Enabled(0.02):Post-TTR and Essential(0.02)
```

Comment (Hekili) [reply TheWarWithin/Priorities/RogueOutlaw.simc]: *Actually*, this is even simpler and seems to work:
> `rtb_buffs_will_lose < buff.loaded_dice.up` | buff.loaded_dice.up & rtb_buffs_will_lose < 3 & ( ! rtb_buffs_will_lose_buff.broadside | buff.broadside.remains < 11 ) & ( ! rtb_buffs_will_lose_buff.ruthless_precision | buff.ruthless_precision.remains < 11 ) & ( ! rtb_buffs_will_lose_buff.true_bearing | buff.true_bearing.remains < 11 )

That is, just removing the parentheses allows the addon to invoke `safenum` here.

**Comment by syrifgit** [comment]
Done @Hekili

File: TheWarWithin/Priorities/RogueOutlaw.simc (hunk 3 above)
Comment (syrifgit) [reply TheWarWithin/Priorities/RogueOutlaw.simc]: Which is easier / closer to simc for doing auto-translation in the future?

**Review by Hekili — APPROVED** [review]

**Pull request merged**